src/picongpu/include/plugins/output/sockets/SocketConnector.hpp

```diff
@@ -109,7 +109,9 @@ class SocketConnector
             size_t zipedSize = zip.compress(tmp + MessageHeader::bytes, ((char*) array) + MessageHeader::bytes, size - MessageHeader::bytes, 6);
             MessageHeader* header = (MessageHeader*) tmp;
             header->data.byte = (uint32_t) zipedSize;
-            write(SocketFD, tmp, zipedSize + MessageHeader::bytes);
+            int nbytes = write(SocketFD, tmp, zipedSize + MessageHeader::bytes);
+            if (nbytes < 0)
+                perror("a socket write error occured");
```

Comment:
I would prefer the C++ method `std::err` over the `C` function call `perror()` .

Please extent the error message with more information: `a socket write error occured in SocketConnector::send`